Tests/PayPalMessagesTests/PayPalMessageModalTests.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import XCTest
+@testable import PayPalMessages
+
+class PayPalMessageModalTests: XCTestCase {
+    let config = PayPalMessageModalConfig(
+        data: .init(
+            clientID: "Test123",
+            environment: .sandbox,
+            amount: 100.0,
+            currency: "USD",
+            offerType: .payLaterLongTerm
+        )
+    )
+
+    var modalViewController: PayPalMessageModal!
+
+    override func setUp() {
+        super.setUp()
+        modalViewController = PayPalMessageModal(config: config)
+        modalViewController.loadViewIfNeeded()
+    }
+
+    override func tearDown() {
+        modalViewController = nil
+        super.tearDown()
+    }
+
+    func testInitialPropertyValues() {
+        let modalViewController = PayPalMessageModal(config: config)
+
+        let supportedOrientations = modalViewController.supportedInterfaceOrientations
+        let preferredOrientation = modalViewController.preferredInterfaceOrientationForPresentation
+        let shouldAutoRotate = modalViewController.shouldAutorotate
+
+        XCTAssertEqual(supportedOrientations, .portrait)
+        XCTAssertEqual(preferredOrientation, .portrait)
+        XCTAssertFalse(shouldAutoRotate)
+    }
+    
+    func testViewDidLoad() {
+        let stateDelegateMock = PayPalMessageModalStateDelegateMock()
+        modalViewController.stateDelegate = stateDelegateMock
+
+        modalViewController.viewDidLoad()
+
+        XCTAssertTrue(stateDelegateMock.onLoadingCalled)
+    }
+    
+    func testViewOnLoadingDelegate() {
+        let stateDelegateMock = PayPalMessageModalStateDelegateMock()
+        modalViewController.stateDelegate = stateDelegateMock
+
+        modalViewController.viewDidLoad()
+
+        XCTAssertTrue(stateDelegateMock.onLoadingCalled)
+    }
```

Comment:
Is this test intended to be a duplicate of the one directly above?

Reply:
Oops, yeah it's a dupe from when I was moving stuff around/renaming. Removed one of them.